### PR TITLE
Make tide easier to run locally and add in commit info.

### PR DESCRIPTION
**Pull request by spxtr**
It still doesn't make any actions yet. That's next!

/area prow
/assign @rmmh

### CODE REVIEW

File: prow/tide/tide.go [hunk 1]
```diff
@@ -103,16 +103,23 @@ func toSimpleState(s kube.ProwJobState) simpleState {
...
-		if smallestNumber < 0 || int(pr.Number) < smallestNumber {
-			smallestNumber = int(pr.Number)
-			smallestPR = pr
+		if smallestNumber != -1 && int(pr.Number) >= smallestNumber {
+			continue
+		}
+		if len(pr.Commits.Nodes) < 1 {
+			continue
 		}
+		if string(pr.Commits.Nodes[0].Commit.Status.State) != "SUCCESS" {
```
Comment (rmmh) [inline prow/tide/tide.go]: `// TODO: match against list of required status checks`

Comment (spxtr) [reply prow/tide/tide.go]: Done.

File: prow/tide/tide.go [hunk 2]
```diff
@@ -170,11 +175,20 @@ func (c *Controller) syncSubpool(ctx context.Context, sp subpool) error {
...
+	}
+	if len(pendings) > 0 {
 		c.log.Info("Do nothing. Waiting for pending PRs.")
-	} else if len(nones) > 0 {
-		c.log.Infof("Trigger tests for PR #%d.", int(pickSmallestNumber(nones).Number))
+		return nil
+	}
+	if len(nones) > 0 {
+		if ok, pr := pickSmallestPassingNumber(nones); ok {
+			c.log.Infof("Trigger tests for PR #%d.", int(pr.Number))
```
Comment (kargakis) [inline prow/tide/tide.go]: Is this meant to rerun broken tests similar to what the [commenter](https://github.com/kubernetes/test-infra/tree/master/experiment/commenter) is doing?

File: prow/tide/tide.go [hunk 3]
```diff
@@ -103,16 +103,24 @@ func toSimpleState(s kube.ProwJobState) simpleState {
...
+			continue
+		}
+		smallestNumber = int(pr.Number)
+		smallestPR = pr
 	}
-	return smallestPR
+	return smallestNumber > -1, smallestPR
 }
 
 // accumulate returns the supplied PRs sorted into three buckets based on their
```
Comment (kargakis) [inline prow/tide/tide.go]: I was rereviewing this helper and I am wondering whether we need to take the PR headref into account when filtering through prowjobs.

File: prow/tide/tide.go (hunk 2 above)
Comment (spxtr) [reply prow/tide/tide.go]: No, it will trigger all jobs on a PR in the tide pool that has passing statuses.

File: prow/tide/tide.go (hunk 3 above)
Comment (spxtr) [reply prow/tide/tide.go]: We already do, in `divideSubpool`.

File: prow/tide/tide.go (hunk 2 above)
Comment (kargakis) [reply prow/tide/tide.go]: Opened https://github.com/kubernetes/test-infra/pull/4270 to discuss more.

File: prow/tide/tide.go (hunk 3 above)
Comment (kargakis) [reply prow/tide/tide.go]: Now I see, thanks.

**Pull request merged**